sdk/identity/azure-identity/tests/test_managed_identity_client.py

```diff
@@ -36,15 +36,15 @@ def test_caching():
         request_factory=lambda _, __: HttpRequest("GET", "http://localhost"), transport=transport
     )
 
-    token = client.get_cached_token(scope)
+    token, refresh_on = client.get_cached_token(scope)
     assert not token
 
     with mock.patch(ManagedIdentityClient.__module__ + ".time.time", lambda: now):
         token = client.request_token(scope)
     assert token.expires_on == expected_expires_on
     assert token.token == expected_token
 
-    token = client.get_cached_token(scope)
+    token, refresh_on = client.get_cached_token(scope)
     assert token.expires_on == expected_expires_on
     assert token.token == expected_token
```

Comment:
Can then also add `assert refresh_on == expected_expires_on` here. Would also be good to check when the expires_in is greater than 2 hours. 

I am thinking we could parametrize this test with something like:

```
@pytest.mark.parametrize("expires_in, expected_refresh_in", ((3600, 3600), (10800, 5400)))
def test_caching(expires_in, expected_refresh_in):
...
...
assert refresh_on == now + expected_refresh_in
```

Reply:
Currently there is no service returns refresh_in property. Let's just do the bare minimal test.